assets/js/universal-dapp.js

```diff
@@ -325,20 +338,32 @@ UniversalDApp.prototype.getCallButton = function(args) {
                 replaceOutput($result, getGasUsedOutput(result));
                 args.appendFunctions(result.createdAddress);
             } else if (self.options.vm){
-                var outputObj;
+                var outputObj = '0x' + result.vm.return.toString('hex');
+                clearOutput($result);
+                $result.append(getReturnOutput(outputObj)).append(getGasUsedOutput(result.vm));
 
-                try {
-                    var outputTypes = [];
-                    for (var i = 0; i < args.abi.outputs.length; i++) {
-                        outputTypes.push(args.abi.outputs[i].type);
+                // Only decode if there supposed to be fields
+                if (args.abi.outputs.length > 0) {
+                    try {
+                        var outputTypes = [];
+                        for (var i = 0; i < args.abi.outputs.length; i++) {
+                            outputTypes.push(args.abi.outputs[i].type);
+                        }
+
+                        // decode data
+                        var decodedObj = EthJS.ABI.rawDecode(null, null, outputTypes, result.vm.return);
+
+                        // format decoded data
+                        decodedObj = EthJS.ABI.stringify(outputTypes, decodedObj);
+                        for (var i = 0; i < outputTypes.length; i++) {
+                            decodedObj[i] = outputTypes[i] + ': ' + decodedObj[i];
```

Comment:
Shouldn't we rather use something like `name: value` or `type name: value` (of course if there is no name, just `type: value`)?